retail/interactive-tutorial/RetailProducts.Samples/ImportProductsBigQueryTableSample.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright 2022 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START retail_import_products_from_big_query]
+// Import products into a catalog from BigQuery table using Retail API
+
+using Google.Cloud.Retail.V2;
+using Google.LongRunning;
+using System;
+
+/// <summary>
+/// The import products BigQuery table sample class.
+/// </summary>
+public class ImportProductsBigQueryTableSample
+{
+    /// <summary>
+    /// Get import products BigQuery request.
+    /// </summary>
+    /// <param name="reconciliationMode">The preffered reconciliation mode.</param>
+    /// <param name="projectId">The current project id.</param>
+    /// <returns>The import product request.</returns>
+    public static ImportProductsRequest GetImportProductsBigQueryRequest(ImportProductsRequest.Types.ReconciliationMode reconciliationMode, string projectId)
+    {
+        string dataSetId = "products";
+        string tableId = "products";
+        // To check error handling use the table of invalid products:
+        // TableId = "products_some_invalid";
+
+        string locationId = "global";
+        string catalogId = "default_catalog";
+        string branchId = "default_branch";
+        BranchName defaultBranch = new BranchName(projectId, locationId, catalogId, branchId);
+        // To check error handling paste the invalid catalog name here:
```

Comment:
Move this above defaultBranch definition

Reply:
done